test/connector.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
```

Comment:
these tests ... aren't great! it's especially hard since it's kind of a sync code path with an async side effect (getCredentials being async makes a lot of the test logic more complex as you see)